src/main/java/org/eclipsefoundation/marketplace/helper/SolrHelper.java

```diff
@@ -0,0 +1,234 @@
+	 * @param c     consumer for setting the value once it is ensured to be a list
+	 * @throws ClassCastException if the type of value consumed is not of type T
+	 */
+	@SuppressWarnings("unchecked")
+	public static <T> void setListField(Object value, Consumer<List<T>> c) {
+		// if value is already a list, set it back immediately and return
+		if (value instanceof List) {
+			c.accept((List<T>) value);
+			return;
+		}
+
+		// creates a new list, adds the object value, and sets it back
+		List<T> l = new ArrayList<>();
+		l.add((T) value);
+		c.accept(l);
+	}
+
+	/**
+	 * Handles repetitive operation of generating a SolrInputField with the given
+	 * key and value, mapping to the passed key in the map.
+	 * 
+	 * @param key    the key for the SolrInputField
+	 * @param value  value for the SolrInputField
+	 * @param fields the map containing the SolrInputFields already generated
+	 */
+	public static void addInputField(String key, Object value, Map<String, SolrInputField> fields) {
+		// Generate the new input field with the set name
+		SolrInputField field = new SolrInputField(key);
+		// boost is used in promoting new content. It is left as default in this call
+		// (1) to not affect ordering of output
+		field.setValue(value, field.getBoost());
+
+		// add the new input value to the map of fields
+		fields.put(key, field);
+	}
+
+	/**
+	 * Calculates the document start index given a positive, non-zero page and page
+	 * size.
+	 * 
+	 * @param page     the page to start from
+	 * @param pageSize the size of result pages
+	 * @return the start index for the given page and page size.
+	 */
+	public static int getDocumentStart(int page, int pageSize) {
+		if (page < 0) {
+			throw new IllegalArgumentException("Cannot generate a document start point with a page less than 0");
+		}
+		if (pageSize < 1) {
+			throw new IllegalArgumentException("Cannot generate a document start point with a pageSize less than 1");
+		}
+
+		return Math.max(0, page - 1) * pageSize;
+	}
+
+	/**
+	 * <p>
+	 * Converts a comma-delimited list of string sort clauses into a form usable by
+	 * Solr. This method takes a string in the following form:
+	 * </p>
+	 * <code>
+	 * [indexed_field_name] [asc|desc], ...
+	 * </code>
+	 * 
+	 * <p>
+	 * e.g. entity_id asc
+	 * </p>
+	 * <p>
+	 * The field name passed into the sort clause must be indexed by Solr, or the
+	 * query will fail. Additionally, if the sort order is not set to either asc or
+	 * desc, the clause will be skipped, allowing the query to continue.
+	 * </p>
+	 * 
+	 * @param sortVal the comma-delimited list of sort clause values
+	 * @return a list of Solr sort clauses
+	 */
+	private static List<SortClause> getSortClauses(String sortVal) {
+		// if empty, return immediately
+		if (StringUtils.isBlank(sortVal)) {
+			return Collections.emptyList();
+		}
+
+		// separate multiple sort clauses by comma
+		String[] sorts = sortVal.split(",");
```

Comment:
String::split is quite dangerous (e.g. what the result spliting of ",," or ", ,, "). For all split, I'd suggest you use https://github.com/google/guava/wiki/StringsExplained#splitter

It's not a blocker for this code, but keep that in mind in the future.

Reply:
Will do!